src/opm/parser/eclipse/EclipseState/IOConfig/IOConfig.cpp

```diff
@@ -190,9 +188,28 @@ namespace Opm {
     std::string IOConfig::getRestartFileName(const std::string& restart_base, int report_step, bool output) const {
         bool unified  = output ? getUNIFOUT() : getUNIFIN();
         bool fmt_file = output ? getFMTOUT()  : getFMTIN();
-        ecl_file_enum file_type = (unified) ? ECL_UNIFIED_RESTART_FILE : ECL_RESTART_FILE;
 
-        return ERT::EclFilename( restart_base , file_type , report_step , fmt_file );
+        auto ext = std::string{};
+        if (unified) {
+            ext = fmt_file ? "FUNRST" : "UNRST";
+        }
+        else {
+            std::ostringstream os;
+
+            const char* fmt_prefix   = "FGH";
+            const char* unfmt_prefix = "XYZ";
+
+            const int cycle = 10 * 1000;
+            const int p_ix  = report_step / cycle;
+            const int n     = report_step % cycle;
+
+            os << (fmt_file ? fmt_prefix[p_ix] : unfmt_prefix[p_ix])
+               << std::setw(4) << std::setfill('0') << n;
```

Comment:
I don't understand this?

Reply:
It creates extensions of the form
```
A0123, B4567, C8901
S0123, T4567, U8901
```
with the letter taken from the prefix.  It is essentially string-stream's version of
```C++
sprintf(buffer, "%c%04d", fmt_file ? fmt_prefix[p_ix] : unfmt_prefix[p_ix], n)
```
without the hazards of `sprintf` (like the missing `static_cast`, for example).